signalfx/client.py

```diff
@@ -106,6 +132,13 @@ def _prepare_api_session(self):
     def _add_to_queue(self, metric_type, datapoint):
         raise NotImplementedError('Subclasses should implement this!')
 
+    def _add_aws_unique_id(self, datapoint):
+        aws_id_dict = {AWS_ID_DIMENSION: self._aws_unique_id}
+        if 'dimensions' in datapoint:
```

Comment:
This might be better to do below and flip
 if datapoint.get('dimensions') is None: